pkg/identityserver/gormstore/end_device_store.go

```diff
@@ -178,6 +179,15 @@ func (s *deviceStore) UpdateEndDevice(ctx context.Context, dev *ttnpb.EndDevice,
 		return nil, err
 	}
 	oldAttributes, oldLocations, oldPicture := devModel.Attributes, devModel.Locations, devModel.Picture
+	if ttnpb.HasAnyField(fieldMask, "last_seen") && dev.GetLastSeen() != nil {
+		if timestamp, err := types.TimestampFromProto(dev.GetLastSeen()); err != nil {
+			return nil, err
+		} else {
+			if timestamp.Before(*devModel.LastSeen) {
+				dev.LastSeen, _ = types.TimestampProto(*devModel.LastSeen)
+			}
+		}
+	}
```

Comment:
Please use `ttnpb.StdTime` and `ttnpb.ProtoTime` so that we have less refactoring to do when we switch away from gogo.